### ISSUE TITLE
[BUG] set-status and record-error not triggered

### ISSUE THREAD

**Issue by Crocmagnon** [jjti/go-spancheck#8]
**Describe the bug**
False negative : reproducing the docs example doesn't trigger the `set-status` and `record-error` checks. `end` reports though.

**To Reproduce**

```go
package main

import (
	"context"
	"errors"
	"go.opentelemetry.io/otel"
)

func subTask() error {
	return errors.New("uh oh")
}

func _() error {
	// span.End is not called on all paths, possible memory leak
	// span.SetStatus is not called on all paths
	// span.RecordError is not called on all paths
	_, span := otel.Tracer("foo").Start(context.Background(), "bar")
	_ = span

	if true {
		// return can be reached without calling span.End
		// return can be reached without calling span.SetStatus
		// return can be reached without calling span.RecordError
		return errors.New("err")
	}

	return nil // return can be reached without calling span.End
}
```
```console
$ spancheck -checks 'end,set-status,record-error' ./...
./main.go:17:5: span.End is not called on all paths, possible memory leak
./main.go:24:9: return can be reached without calling span.End
```

**Expected behavior**
All three checks should report issues.

**Comment by jjti** [comment]
Hey @Crocmagnon , thank you for reporting this! In preparing for the golangci-lint merge, I got lazy and introduced some bugs with the CLI. Fixing now. Eg of fixed cli for your snippet above

```bash
$ spancheck -checks "end,set-status,record-error" ./testdata/enableall/enable_all.go
/Users/josh/GitHub/jjti/go-spancheck/testdata/enableall/enable_all.go:17:2: span.End is not called on all paths, possible memory leak
/Users/josh/GitHub/jjti/go-spancheck/testdata/enableall/enable_all.go:21:3: return can be reached without calling span.End
/Users/josh/GitHub/jjti/go-spancheck/testdata/enableall/enable_all.go:17:2: span.SetStatus is not called on all paths
/Users/josh/GitHub/jjti/go-spancheck/testdata/enableall/enable_all.go:21:3: return can be reached without calling span.SetStatus
/Users/josh/GitHub/jjti/go-spancheck/testdata/enableall/enable_all.go:17:2: span.RecordError is not called on all paths
/Users/josh/GitHub/jjti/go-spancheck/testdata/enableall/enable_all.go:21:3: return can be reached without calling span.RecordError
```

**Comment by jjti** [comment]
@Crocmagnon I think this was fixed, here: https://github.com/jjti/go-spancheck/releases/tag/v0.4.3

Closing but please re-open if you hit the issue with >= 0.4.3! Thank you again for reporting

**Comment by Crocmagnon** [comment]
I can confirm the checks now properly report issues.
